pkg/yurthub/multiplexer/storage/store.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"math/rand"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/storage"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+)
+
+const minWatchRequestSeconds = 300
+
+var ErrNoSupport = errors.New("Don't Support Method ")
+
+type store struct {
+	restClient rest.Interface
+	resource   string
+}
+
+func NewStore(restClient rest.Interface, resource string) *store {
+	return &store{
+		restClient: restClient,
+		resource:   resource,
+	}
+}
+
+func (rs *store) GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object) error {
```

Comment:
paramter `key` is unused?

Reply:
Yes, unlike retrieving data from etcd, this store requests the apiserver to obtain all resources without needing this key to specify the storage path for resource retrieval. In order to reuse the cache in the apiserver (https://github.com/kubernetes/apiserver/blob/master/pkg/storage/cacher/cacher.go#L74), the store needs to retain this parameter.